app/src/main/java/org/fossasia/openevent/app/data/models/Ticket.java

```diff
@@ -19,7 +20,7 @@
     private long quantity;
     private String type;
 
-    @ForeignKey(stubbedRelationship = true)
+    @ForeignKey(stubbedRelationship = true, onDelete = ForeignKeyAction.CASCADE)
```

Comment:
on delete is a trigger which works on foreign keys. Meaning if this record containing this foreign key is deleted, what do you want me to do with the foreign key model, CASCADE means delete that too. Meaning, it is saying if the event foreign key is deleted (when ticket will be deleted), delete the event too. You can actually try by calling ticket.delete(). 

The bug is that this constraint is only satisfied if you call model.delete(), not the model adaper.delete(model)

Deleting event will work fine obviously, test that is deleting a ticket also is deleting event or not. But, it won't delete now because of this bug, and cause trouble later when it is fixed.